packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -318,10 +318,13 @@ class _ProductListPageState extends State<ProductListPage>
         onDismissed: (final DismissDirection direction) async {
           final bool removed = productList.remove(barcode);
           if (removed) {
-            await DaoProductList(localDatabase).put(productList);
             _selectedBarcodes.remove(barcode);
-            setState(() => barcodes.removeAt(index));
+            await DaoProductList(localDatabase).put(productList);
           }
+          final String removedItem = barcodes.removeAt(index);
+          setState(() {});
+          // ignore: use_build_context_synchronously
```

Comment:
Please use instead:
```dart
if (!mounted) {
  return;
}
```

Or put in a variable beforehand `ScaffoldMessenger.of(context)` and reuse it.

Reply:
Still to be done.